lib/internal/bootstrap_node.js

```diff
@@ -114,6 +114,10 @@
       }
 
       if (process._eval != null && !process._forceRepl) {
+        if (process._syntax_check_only != null) {
+          console.error('--check and --eval flags are mutually exclusive.');
+          process.exit(1);
```

Comment:
I couldn't find any other logic like this for flags, so I sort of improvised with the error message. Let me know if there's a better way to do this.